core/src/apps/cardano/sign_tx.py

```diff
@@ -778,6 +799,11 @@ def _validate_witness(
         _ensure_no_payment_witness(witness)
 
 
+def _validate_script_data_hash(script_data_hash: bytes) -> None:
```

Comment:
Please add a TODO comment at the end of this function which would remind us to re-assess the validation once more - specifically with regards to signing modes.